firebase_admin/__init__.py

```diff
@@ -186,3 +197,41 @@ def credential(self):
     @property
     def options(self):
         return self._options
+
+    def get_token(self):
+        """Returns an OAuth2 bearer token.
+
+           This method may return a cached token. But it handles cache invalidation, and therefore
+           is guaranteed to always return unexpired tokens.
+
+           Returns:
+             string: An unexpired OAuth2 token.
+        """
+        if not self._token_valid():
+            self._token = self._credential.get_access_token()
+        return self._token.access_token
+
+    def _token_valid(self):
+        if self._token is None:
+            return False
+        skewed_expiry = self._token.expiry - datetime.timedelta(seconds=_CLOCK_SKEW_SECONDS)
+        return _clock() < skewed_expiry
+
+    def _get_service(self, name, initializer):
+        if not name or not isinstance(name, six.string_types):
+            raise ValueError(
+                'Illegal name argument: "{0}". Name must be a non-empty string.'.format(name))
+        with self._lock:
+            if self._services is None:
+                raise ValueError(
+                    'Service requested from deleted Firebase App: "{0}".'.format(self._name))
+            if name not in self._services:
+                self._services[name] = initializer(self)
+            return self._services[name]
+
+    def _cleanup(self):
+        with self._lock:
+            for service in self._services:
+                if hasattr(service, 'close'):
+                    service.close()
```

Comment:
It looks like `callable()` is not available in Python 3.0 and 3.1. They dropped it and added it back in 3.2: https://docs.python.org/3/library/functions.html#callable

So I implemented this extra check by looking for the `__call__` property on the function itself.

Also found a bug in the logic while testing it. I've fixed it and updated the tests to check for that.